rfcs/text/0000-contributed-pipelines.md

```diff
@@ -0,0 +1,161 @@
+2. Upload the WDL to the public Terra tools repository, with configurations, data, and descriptions for each mode the pipeline can be run in, and import the tool into a workspace.
+    3. Another option is to put the tool into Dockstore and then link it to Terra.
+3. Upload small testing data, necessary references, and any benchmarking datasets to the workspace. Ensure these data are eligible for public, open use.
+4. Write a markdown formatted workspace description that summarizes:
+    1. the data being analyzed, 
+    2. the way the input data is generated,
+    3. the computational stages of the pipeline,
+    4. the output data,
+    5. and how the pipeline meets the scientific and technical contribution standards
+5. Run the pipeline in the workspace, in each mode, with at least the test data, so that the outputs can be verified. If there are data that work with your pipeline in the HCA DCP, demonstrate analyzing this data with your pipeline in the workspace.
+6. Using another wdl tool, write a checker test that verifies the outputs of the pipeline meet the technical and scientific expectations for each run mode of the pipeline.
+7. Share the workspace with write access to pipelines-team@data.humancellatlas.org for internal review by the DCP pipelines team. The workspace will then be announced to the DCP and HCA community for review.
+    1. During review, we may request instructions and code to read the output file into a common scientific computing language as a sparse or dense array. 
+8. Respond to the community and DCP concerns, and update the submission as requested.
+9. When the pipeline has met the criteria for consideration and has been approved by the DCP and community, the DCP pipelines team will tag the workspace "HCA-contributed-pipeline". The DCP will then determine when it can be prioritized to be pulled into production, and will communicate the timeline and expectations with the contributor and community (see Section 3.3, Prioritizing Contributions, below).
+10. Running pipelines in the cloud and in Terra requires a google cloud billing project. If you have interest in contributing and have difficulty obtaining a google cloud billing project, please reach out via the above email.
+
+#### Why Terra?
+Contributing pipelines via Terra has a few benefits:
+1. Ensures the pipeline can be run in the DCP pipeline infrastructure
+2. Provides support for contributors developing and testing the pipelines
+3. Makes the pipelines immediately available to the community, where users can run the pipelines on HCA data, in the case where they may be useful but not immediately able to be accepted to the DCP, for example if they haven’t met an acceptance criteria, or aren’t prioritized into production because of a reason in section 3.3
```

Comment:
If the pipeline is public in Terra it's accessible to anyone, and anyone could make an account if they wanted to clone and run it. If the person making the pipeline is savvy enough, they could submit it through Dockstore and then show it running in Terra. If you'd like, we could add that information to this process?

Reply:
There was some contention among the authors on this point. An alternative proposal for contribution took the following form: 

1. The DCP team sources contributions by opening issues for specific assays. Community members can also open issues to discuss potential contributions that are not sought by the team or on the roadmap. 
2. Contribution occurs via PR to humancellatlas/skylab
3. Contribution require users to write tests that match and pass skylab CI, which requires (i) construction of docker containers, (ii) specification of test inputs and (iii) specification of expected results

Some considerations:
1. This requires contributors to understand the github issue>fork>pr open source contribution process, which may have both positive and negative impacts
2. Users need not have access to google cloud resources, because they are provided by skylab CI. 
3. Contributors are rewarded for their efforts with commits recorded directly against the DCP pipelines code, motivating direct scientific collaboration on our pipelines product.

Reply:
I really like the more targeted approach of "DCP...opening issues for specific assays". I think this also will invite more community input & discussion, which will result in fruitful scientific discussions around the pipelines, especially when 2+ are proposed, and ultimately community buy-in.